api-gateway/src/main/java/com/petclinic/bffapigateway/domainclientlayer/CustomersServiceClient.java

```diff
@@ -162,16 +162,12 @@ public Mono<OwnerResponseDTO> createOwner(OwnerRequestDTO model) {
     }
 
     public Mono<OwnerResponseDTO> addOwner(Mono<OwnerRequestDTO> model) {
-        String ownerId = UUID.randomUUID().toString();
-        return model.flatMap(requestDTO -> {
-            requestDTO.setOwnerId(ownerId);
-            return webClientBuilder.build()
-                    .post()
-                    .uri(customersServiceUrl + "/owners")
-                    .body(BodyInserters.fromValue(requestDTO))
-                    .retrieve()
-                    .bodyToMono(OwnerResponseDTO.class);
-        });
+        return model.flatMap(requestDTO -> webClientBuilder.build()
+                .post()
+                .uri(customersServiceUrl + "/owners")
+                .body(BodyInserters.fromValue(requestDTO))
+                .retrieve()
+                .bodyToMono(OwnerResponseDTO.class));
```

Comment:
Thank you I did this and I see what the problem is thanks for removing the redundant UUID generation for ownerId and explaining it to me

Reply:
No problem!